core/include/drivers/clk.h

```diff
@@ -6,60 +6,116 @@
 #ifndef __DRIVERS_CLK_H
 #define __DRIVERS_CLK_H
 
+#include <assert.h>
 #include <kernel/refcount.h>
 #include <stdint.h>
 #include <tee_api_types.h>
+#include <util.h>
 
 /* Flags for clock */
 #define CLK_SET_RATE_GATE	BIT(0) /* must be gated across rate change */
 #define CLK_SET_PARENT_GATE	BIT(1) /* must be gated across re-parent */
 
+/*
+ * Type of clock instantiated:
+ *
+ * CLK_OPS_ELT identities a full fledged clock. The struct clk * reference can be
+ * cast to struct clk_elt * to access clock element data.
+ *
+ * CLK_OPS_LIGHTWEIGHT identifies a simple clock, no parents, flags and
+ * features handle by strcut clk_elt instances to optimize struct clk
+ * memory footprint.
+ */
+enum clk_ops_id {
+	CLK_OPS_INVALID = 0,
+	CLK_OPS_ELT,
+	CLK_OPS_LIGHTWEIGHT,
```

Comment:
element is shortened in `ELT` but lightweight is kept entirely. Either use ELT/LW or ELEMENT/LIGHTWEIGHT

Reply:
Ok. Better change to explicit ELEMENT (or GENERIC or ...) and LIGHTWEIGHT for `CLK_OPS_*`